### PR TITLE
Working on cookies

### PR DESCRIPTION
**Pull request by zRalphy**

### CODE REVIEW

File: todo/src/main/java/org/openapitools/controller/UserApiController.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		this.userService = userService;
+	}
+
+	@Override
+	public Optional<NativeWebRequest> getRequest() {
+		return Optional.ofNullable(request);
+	}
+
+	@Override
+	public ResponseEntity<Void> login(UserLoginRequest userLoginRequest) throws ApiException {
```
Comment (PawelAdamski) [inline todo/src/main/java/org/openapitools/controller/UserApiController.java]: We don't need login endpoint, we use filter instead.

File: todo/src/main/java/org/openapitools/controller/UserApiController.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	@Override
+	public ResponseEntity<String> getUserInfo() throws ApiException{
+		Principal token = request.getUserPrincipal();
+		String userName;
+		if(token instanceof OAuth2AuthenticationToken){
+			OAuth2AuthenticationToken defaultToken = (OAuth2AuthenticationToken) request.getUserPrincipal();
+			 userName = defaultToken.getPrincipal().getAttributes().get("name").toString();
+			return new ResponseEntity<> (userName, HttpStatus.OK);
+		} else {
+			throw new ApiException(HttpStatus.FORBIDDEN.value());
```
Comment (PawelAdamski) [inline todo/src/main/java/org/openapitools/controller/UserApiController.java:47]: You can implement verification if user has logged in using OAuth or not using roles. 
Please assign a role "OAuth_User" to users that have logged in via Google. When you have this you can change security config for UserInfo endpoint to `.requestMatchers("/user/info").hasRole("OAUTH_USER")`

**Review by PawelAdamski — CHANGES_REQUESTED** [review]

File: todo/src/main/java/org/openapitools/controller/UserApiController.java (hunk 1 above)
Comment (zRalphy) [reply todo/src/main/java/org/openapitools/controller/UserApiController.java]: Right

File: todo/src/main/java/org/openapitools/controller/UserApiController.java (hunk 2 above)
Comment (zRalphy) [reply todo/src/main/java/org/openapitools/controller/UserApiController.java:47]: Done

**Review by PawelAdamski — APPROVED** [review]
Looks good to me.

**Pull request merged**